asset/swift.go

```diff
@@ -0,0 +1,67 @@
+package asset
+
+import (
+	"bytes"
+	"crypto/md5"
+	"encoding/hex"
+	"io"
+	"io/ioutil"
+
+	"github.com/cozy/swift"
+)
+
+type SwiftFS struct {
+	Connection *swift.Connection
+}
+
+func (s *SwiftFS) AddAsset(asset *GlobalAsset, content io.Reader) error {
+	// Creating object in swift
+	sc := s.Connection
+
+	buf, err := ioutil.ReadAll(content)
+	if err != nil {
+		return err
+	}
+
+	// Calculating md5sum for swift insertion
+	md5sum := md5.New()
+	_, err = md5sum.Write(buf)
+	if err != nil {
+		return err
+	}
+	sum := md5sum.Sum(nil)
+
+	f, err := sc.ObjectCreate(AssetContainerName, asset.Shasum, true, hex.EncodeToString(sum), asset.ContentType, nil)
```

Comment:
We didn't understand each other. I didn't want that we compute the md5sum before calling `ObjectCreate`, just use `checkHash=true`, like this:

```go
f, err := sc.ObjectCreate(AssetContainerName, asset.Shasum, true, "", asset.ContentType, nil)
```